### PR TITLE
Added workflow for building & publishing site

### PR DESCRIPTION
**Pull request by MukulMaheshwari**
Auto deployment of layer5-ng to github pages via actions #3

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+      - name: Install dependencies & build
+        run: |
+          apt install nodejs -y
+          curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | sudo apt-key add -
+          echo "deb https://dl.yarnpkg.com/debian/ stable main" | sudo tee /etc/apt/sources.list.d/yarn.list
+          apt update && apt install yarn
+          npm install -g gatsby-cli
+          yarn global add gatsby-dev-cli
+          npm install
```
Comment (Tanuj22) [inline .github/workflows/ci.yml:23]: Just a heads up. `npm install` should be run inside the `site` directory

**Comment by leecalcote** [comment]
> @MukulMaheshwari You may need to follow https://github.com/layer5io/layer5-ng/pull/15/checks?check_run_id=841304155 to pass the DCO checks as you haven't signed the PR.

@MukulMaheshwari will you execute these steps?

**Comment by MukulMaheshwari** [comment]
Sure, I was working on caching & will follow these steps in next commit.

**Comment by leecalcote** [comment]
@MukulMaheshwari 👍

**Comment by kushthedude** [comment]
@MukulMaheshwari Can you update this PR to the latest workflow you shared?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - run: cd site
+      - uses: enriikke/gatsby-gh-pages-action@v2
+        with:
+          access-token: ${{ secrets.ACCESS_TOKEN }}
```
Comment (leecalcote) [inline .github/workflows/ci.yml:16]: Looking good, @MukulMaheshwari.

Will you change the secret name to `GH_ACCESS_TOKEN`?

Comment (MukulMaheshwari) [reply .github/workflows/ci.yml:16]: Sure

**Pull request merged**